### ISSUE TITLE
What GPU is required to run it ?

### ISSUE THREAD

**Issue by Eddh** [lucidrains/big-sleep#3]
I tried executing the dream command on my laptop computer with a Quadro P2000 with 4 GB of vram. I got a CUDA out of memory error.

**Comment by lucidrains** [comment]
@Eddh you'll need a graphics card with at least 9 GB of ram!

**Comment by enricoros** [comment]
@Eddh the minimum I ran this on is a 2070 SUPER: 8GB ram which is all used

**Comment by lucidrains** [comment]
@remdu Just use the new colab link if you don't have the processing power :)

**Comment by n00mkrad** [comment]
An 8 GB GPU seems to work if it's a second GPU so its base VRAM usage is 0.0 GB.

**Comment by iGio90** [comment]
Do you think there is any param that can be tuned to run on some lower, yet quite powerfull (i think), gpu?

```
RuntimeError: CUDA out of memory. Tried to allocate 128.00 MiB (GPU 0; 6.00 GiB total capacity; 3.97 GiB already allocated; 117.08 MiB free; 3.99 GiB reserved in total by PyTorch)
```

from what I understood all around this depends from the batch size.. so sad I can't run this (even by waiting some more minutes)

**Comment by enricoros** [comment]
@iGio90 can you try with image_size=128 ? may pull a smaller model.. not sure it will work, but try that

**Comment by iGio90** [comment]
Yeah, I tried reducing to 128 but I had the same memory error

**Comment by enricoros** [comment]
@iGio90 I feel you, these days Video RAM is very important to run models. There's a kind of assumption on the research side that the card should have > 11GB RAM.. models converge to that runtime size.

**Comment by chozabu** [comment]
This can be made to run on CPU, but on a i7-6700 it took ~10 hrs to generate an image

**Comment by yangboz** [comment]
> This can be made to run on CPU, but on a i7-6700 it took ~10 hrs to generate an image

nope, it will throw "assert torch.cuda.is_available(), 'CUDA must be available in order to use Deep Daze'"

**Comment by LtqxWYEG** [comment]
Damn. You all say: "A GPU with 8GB VRAM is required" and "Everyone with A GPU". 
I have an AMD RX570 with 8GBN VRAM. Imagine how I feel suddenly realizing after installation that this only works with CUDA. I'm so sad and mad... I can't do any cool AI stuff with my card. FFS

**Comment by n00mkrad** [comment]
> Damn. You all say: "A GPU with 8GB VRAM is required" and "Everyone with A GPU".
> I have an AMD RX570 with 8GBN VRAM. Imagine how I feel suddenly realizing after installation that this only works with CUDA. I'm so sad and mad... I can't do any cool AI stuff with my card. FFS Edit: Mad at me for buying amd

Use Colab or ROCm.

**Comment by LtqxWYEG** [comment]
Would that work with this and others programs?

**Comment by n00mkrad** [comment]
> Would that work with this and others programs? How easy to implement is that?
> Edit: ROCm is only for Linux and Colab is not using my hardware, right? So it's probably really slow. :(
> EDIT: Darn. I just saw that there are pre-made colab notebooks. I made my own. Well whatever. It seems to work, but I don't know HOW slow it is. How slow is 1.41 it/s?

Colab using a P100 is definitely much faster than your RX 570, and you have twice the VRAM.

**Comment by LtqxWYEG** [comment]
> Colab using a P100 is definitely much faster than your RX 570, and you have twice the VRAM.

This made me read through things and apparently it uses either one of these and you can't choose: Nvidia K80s, T4s, P4s and P100s. And there are **un**certain limits to GPU and VM usage. So it can just suddenly stop running on GPU with no warning. 
There's also way to appear to be local notebooks, which would use my hardware, that run via Jupyter - but of course, I don't have an nVidia card.

**Comment by drewnydrzew** [comment]
HI, I am considering buying nvidia quadro m4000 or m6000. Will it work with big sleep?

**Comment by wolfgangmeyers** [comment]
> HI, I am considering buying nvidia quadro m4000 or m6000. Will it work with big sleep?

From what I've seen, the dedicated VRAM is the constraint. I'm using a rtx 3070 with 8GB dedicated VRAM. It is barely enough VRAM to run Big Sleep, but the actual GPU processing power is barely used.

I've been able to get VRAM usage down near 6GB and even 4GB by lowering image_size and num_cutouts parameters.
--num-cutouts=16 and --image-size=128 should work on a 4GB card, but I haven't tested yet.

Looks like the m4000 has 8GB of VRAM and the m6000 has 24GB (!!) so either should work.

**Comment by areyougood** [comment]
can i use shared GPU memory instead of dedicated? if so, how?

**Comment by georgbuechner** [comment]
Just to make this clear: is there any way to make this work with an AMD (Radeon 5700) card?

**Comment by LtqxWYEG** [comment]
I don't REALLY know, but you can always use the colab notebook.
I forked the notebook with better description and more settings: https://colab.research.google.com/drive/1zVHK4t3nXQTsu5AskOOOf3Mc9TnhltUO?usp=sharing
